File: testScripts/pwmTest.c
/*
 Change the Intensity of LED using PWM on Raspberry Pi
 http://www.electronicwings.com
 */

#include <wiringPi.h>

#include <stdio.h>
#include <stdlib.h>

const int PWM_pin = 1;   /* GPIO 1 as per WiringPi, GPIO18 as per BCM */

int main (void)
{
  int intensity ;
  int pwmVal = 500;            

  if (wiringPiSetup () == -1)
    exit (1) ;

  pinMode (PWM_pin, PWM_OUTPUT) ; /* set PWM pin as output */

  while (1)
  {
	
    for (intensity = 500 ; intensity < 502 ; ++intensity)
    {
      pwmWrite (PWM_pin, intensity) ;	/* provide PWM value for duty cycle */
      delay (1) ;
    }
    delay(1);

    for (intensity = 501 ; intensity >= 500 ; --intensity)
    {
      pwmWrite (PWM_pin, intensity) ;
      delay (1) ;
    }
    delay(1);
}
}

File: testScripts/pwmTest.py
import wiringpi, time

def reset_pins(io):
    io.pinMode(26,io.OUTPUT)
    io.digitalWrite(26, io.LOW)


def pwm_dimm(io):
    STEP = 8
    pin = 26 # only supported on this pin
    io.pinMode(pin,io.PWM_OUTPUT)
    io.pwmWrite(pin, 500)


io = wiringpi.GPIO(wiringpi.GPIO.WPI_MODE_PINS)

try:
    reset_pins(io)    
    pwm_dimm(io)
    time.sleep(5)

except (KeyboardInterrupt, SystemExit):
    reset_pins(io)
